api/src/test/java/org/openmrs/CohortMembershipTest.java

```diff
@@ -252,4 +252,147 @@ public void compareTo_shouldSort() throws Exception {
 		
 		assertThat(list, contains(startedRecently, startedLongAgo, noStartOrEnd, endedRecently, endedLongAgo, voided));
 	}
+	
+	@Test
+	public void equal_shouldReturnTrueIfObjectReferenceAreTheSame() {
+		CohortMembership cohortMembership = new CohortMembership(12);
+		
+		assertTrue(cohortMembership.equals(cohortMembership));
+	
+	}
+	
+	@Test
+	public void equal_shouldReturnFalseIfObjectIsNull() {
+		// A null argument to equal should not lead to an exception
+		CohortMembership cohortMembership = new CohortMembership(12);
+		
+		assertFalse(cohortMembership.equals(null));
+		
+	}
+	
+	@Test
+	public void equal_shouldReturnTrueIfStartDateEndDatePatientAreSame() {
+		
+		// Creating two logical identical CohortMemberships to test equal method
+		CohortMembership cohortMembershipOne = new CohortMembership(12);
+		CohortMembership cohortMembershipTwo = new CohortMembership(12);
+		
+		Date startDate = new Date(1545140935);
+		Date endDate = new Date(1577836800);
+		
+		cohortMembershipOne.setEndDate(endDate);
+		cohortMembershipOne.setStartDate(startDate);
+		cohortMembershipTwo.setStartDate(startDate);
+		cohortMembershipTwo.setEndDate(endDate);
+		
+		assertTrue(cohortMembershipOne.equals(cohortMembershipTwo));
+		
+	}
+	
+	@Test
+	public void equal_shouldReturnFalseIfStartDateEndDatePatientAreNotSame() {
+		
+		// Creating two logical identical CohortMemberships ot test equal
+		CohortMembership cohortMembershipOne = new CohortMembership(13);
+		CohortMembership cohortMembershipTwo = new CohortMembership(12);
+		
+		Date startDate1 = new Date(1545140935);
+		Date endDate1 = new Date(1577836801);
+		
+		Date startDate2 = new Date(1545140935);
+		Date endDate2 = new Date(1577836800);
+		
+		
+		cohortMembershipOne.setEndDate(endDate1);
+		cohortMembershipOne.setStartDate(startDate1);
+		cohortMembershipTwo.setStartDate(startDate2);
+		cohortMembershipTwo.setEndDate(endDate2);
+		
+		assertFalse(cohortMembershipOne.equals(cohortMembershipTwo));
+	}
+	
+	@Test
+	public void equal_shouldReturnFalseIfOnlyStartDateOfOneObjectIsNull() {
+		
+		CohortMembership cohortMembershipOne = new CohortMembership(13);
+		CohortMembership cohortMembershipTwo = new CohortMembership(12);
+		
+		Date startDate1 = null;
+		Date endDate1 = new Date(1577836801);
+		
+		Date startDate2 = new Date(1545140935);
+		Date endDate2 = null;
+		
+		
+		cohortMembershipOne.setEndDate(endDate1);
+		cohortMembershipOne.setStartDate(startDate1);
+		cohortMembershipTwo.setStartDate(startDate2);
+		cohortMembershipTwo.setEndDate(endDate2);
+		
+		assertFalse(cohortMembershipOne.equals(cohortMembershipTwo));
+	}
+	
+	@Test
+	public void equal_shouldReturnFalseIfOnlyEndDateOfOneObjectIsNull() {
```

Comment:
end date of one object or two objects?

Reply:
Secondly, would it return true of the patient ids were the same?

Reply:
(1) Only EndDate of one object is null. That is why both objects are not equal since the EndDate is different
(2) I fixed it. It was actually a mistake of me